### PR TITLE
Remove chrome artifacts

### PR DESCRIPTION
**Pull request by regiskuckaertz**
There was some redundancy across several modules regarding an unwanted behaviour in Chrome. I moved the `removeChromeArtifacts` over to the node helpers and rewrote the affected parts. Additionally, I made sure `SPAN`s were removed only if they had no attributes.

### CODE REVIEW

File: src/node.js [hunk 1]
```diff
@@ -14,6 +14,16 @@ define([
     return blockElementNames.includes(node.nodeName);
   }
 
+  // Source: https://developer.mozilla.org/en-US/docs/Web/HTML/Inline_elemente
+  var inlineElementNames = Immutable.Set.of('B', 'BIG', 'I', 'SMALL', 'TT',
```
Comment (rrees) [inline src/node.js]: This might be better in a constants-style module

File: src/node.js [hunk 2]
```diff
@@ -105,7 +115,53 @@ define([
...
+      parentElement,
+      NodeFilter.SHOW_ELEMENT,
+      // arguments to .bind(), starting from the second, are automatically
+      // prepended to the actual arguments when the function is called
+      isInlineWithStyle.bind(null, window.getComputedStyle(parentElement))
+    );
+    var emptySpans = Immutable.List();
+    var node;
+
+    while (node = iterator.nextNode()) {
```
Comment (rrees) [inline src/node.js]: Could this work as filter? I am assuming not.

File: src/node.js [hunk 3]
```diff
@@ -105,7 +115,53 @@ define([
...
+      node.style.lineHeight = null;
+      if (!node.getAttribute('style')) {
+        node.removeAttribute('style');
+      }
+      if (node.nodeName === 'SPAN' && node.attributes.length === 0) {
+        emptySpans = emptySpans.push(node);
+      }
+    }
+
+    while (!!emptySpans.size) {
```
Comment (rrees) [inline src/node.js]: You have isEmpty on Immutable.List but it would probably be better as a forEach to make clear it is using side-effects.

I realise this might be less efficient but is clearer.

File: src/plugins/core/patches/commands/insert-list.js [hunk 4]
```diff
@@ -1,4 +1,4 @@
-define([], function () {
```
Comment (rrees) [inline src/plugins/core/patches/commands/insert-list.js]: There might have been an obscure module bug that meant we kept the empty arrays. Could you submit this change in a different PR so that we don't have yank the consolidation because of the loader.

**Comment by rrees** [comment]
Overall looks sound to me :+1:, would value a second opinion from the team though.

File: src/node.js (hunk 1 above)
Comment (regiskuckaertz) [inline src/node.js]: Yes, sure! I assume there will be others with time (like `blockElementNames` for instance), what do you think we put them in a folder, say `/src/vars` or `/src/node/vars`?

File: src/node.js (hunk 2 above)
Comment (regiskuckaertz) [inline src/node.js]: Oh yeah, I think it could, something like:

```
var nodes = Immutable.List(parentElement.querySelectorAll(inlineElementNames.join(',')));
nodes = nodes.filter(...);
```

Ideally the selector would select only nodes with a `style` attribute, so this would need to be tweaked. Your call.

File: src/node.js (hunk 3 above)
Comment (regiskuckaertz) [inline src/node.js]: :+1:

File: src/node.js (hunk 2 above)
Comment (rrees) [inline src/node.js]: Yeah, I think this would be more elegant.

**Comment by regiskuckaertz** [comment]
There it is, indeed it's nicer. Thanks for the suggestion @rrees

**Comment by rrees** [comment]
Small (last) thing could the module be called constants rather than vars. Because they are actually invariants.

**Comment by regiskuckaertz** [comment]
No problemo, I'll take care of this tonight :+1:
